Source/Core/DiscIO/VolumeWii.cpp

```diff
@@ -394,10 +394,11 @@ bool VolumeWii::CheckIntegrity(const Partition& partition) const
     return false;
 
   // Get partition data size
-  u32 part_size_div4;
-  m_reader->Read(partition.offset + 0x2BC, sizeof(u32), reinterpret_cast<u8*>(&part_size_div4));
-  const u64 part_data_size = static_cast<u64>(Common::swap32(part_size_div4)) * 4;
+  const auto part_size_div4 = ReadSwapped<u32>(partition.offset + 0x2BC, PARTITION_NONE);
+  if (!part_size_div4)
+    return false;
 
+  const u32 part_data_size = part_size_div4.value() * 4;
```

Comment:
`ReadSwapped` handles the multiplication by 4 for you, so don't do it manually here.

Reply:
Sorry, I'm mistaken. If you use `ReadSwappedAndShifted` instead, it handles the multiplication by 4 for you.